src/ProjectsExport.php

```diff
@@ -0,0 +1,118 @@
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace GlpiPlugin\Gantt;
+
+class ProjectsExport
+{
+    private int $project_id;
+
+    public function __construct(int $project_id) {
+        $this->project_id = $project_id;
+    }
+
+    private function loadProjectsData(): array
+    {
+        $items = [];
+        $factory = new DataFactory();
+        $factory->getItemsForProject($items, $this->project_id);
+
+        $id_refs = [];
+        $to_export = [];
+        foreach ($items as $item) {
+            switch ($item->type) {
+                case 'project':
+                    $project = [
+                        'id' => $item->id,
+                        'name' => $item->text,
+                        'start_date' => $item->start_date,
+                        'end_date' => $item->end_date,
+                    ];
+
+                    $id_refs[$item->id] = array_merge(
+                        $id_refs[$item->id] ?? [],
+                        $project
+                    );
+                    if ($item->parent == 0) {
+                        $to_export[] = &$id_refs[$item->id];
+                    } else {
+                        $id_refs[$item->parent]['projects'][] = &$id_refs[$item->id];
+                    }
+                    break;
+                case 'task':
+                    $task = [
+                        'id' => $item->linktask_id,
+                        'name' => $item->text,
+                        'start_date' => $item->start_date,
+                        'end_date' => $item->end_date,
+                        'progress' => $item->progress,
+                    ];
+
+                    $id_refs[$item->id] = array_merge(
+                        $id_refs[$item->id] ?? [],
+                        $task
+                    );
+                    $id_refs[$item->parent]['tasks'][] = &$id_refs[$item->id];
+                    break;
+                case 'milestone':
+                    $milestone = [
+                        'id' => $item->linktask_id,
+                        'name' => $item->text,
+                        'date' => $item->start_date,
+                    ];
+
+                    $id_refs[$item->id] = array_merge(
+                        $id_refs[$item->id] ?? [],
+                        $milestone
+                    );
+                    $id_refs[$item->parent]['milestones'][] = &$id_refs[$item->id];
+                    break;
+            }
+        }
+
+        return $to_export;
+    }
+
+    public function json(): string
+    {
+        header_remove('Content-Type');
+        header_remove('Content-Disposition');
+
+        header('Content-Type: application/json');
+        header('Content-Disposition: attachment; filename="gantt-data.json"');
+
+        return json_encode($this->loadProjectsData());
```

Comment:
We usually try to define the headers in the `front` file.
Can you move the header to the front file and just keep the `return json_encode` part here ?

`header_remove` shouldn't be needed too, did you encounter some issues that forced you to use it ?

Reply:
I've added `header_remove` as a preventive measure to ensure that the headers are good, but in the end they're useless as I'm writing over them again. I've had no problems without them.